### DISCUSSION TITLE
Support for OIDC4VC standards

### DISCUSSION THREAD

**Discussion by piraces** [zitadel/zitadel#7133] [Ideas]
OpenID Foundation is developing a set of standards for handling verifiable credentials. They are being developed under the name [OIDC4VC - OpenId Connect for Verifiable Credentials](https://openid.net/openid4vc/), and they include standards for [issuing](https://openid.bitbucket.io/connect/openid-4-verifiable-credential-issuance-1_0.html) and [presenting credentials](https://openid.bitbucket.io/connect/openid-4-verifiable-presentations-1_0.html) under what is called the issuer-holder-verifier model, which attempts to avoid IDPs auditing every authentication by users.

Those standars are still emergent, but they are likely to get great adoption due to being required for certain use cases by EU eIDAS v2: [European Digital Identity Architecture and Reference Framework](https://github.com/eu-digital-identity-wallet/architecture-and-reference-framework).

Since zitadel implements other standards from the OpenID Foundation and its certified, facilitating the integration of systems accepting those novel authentication and attestation exchange methods would be ideal.

## Related work
There are already some [implementations listed here](https://openid.net/openid4vc/). Microsoft is also working in a set of services under its [Microsoft Entra Verified Id](https://www.microsoft.com/en-en/security/business/identity-access/microsoft-entra-verified-id) "product".

In the [Ory Hydra](https://github.com/ory/hydra) project they have also [made some related developments](https://www.ory.sh/docs/hydra/reference/api#tag/oidc/operation/createVerifiableCredential) and plan to keep going to support this novel standards.

In the demos mentioned above, novel standards are integrated with existing IAMs (Keycloak) via OIDC standard methods, and new services are in charge of issuing and verifying credentials according to novel standards. But that adds an integration step that could be avoided if OIDC implementors (isolated or included in IDPs) implement and accept those standards.

**Comment by hifabienne** [comment]
hei @livio-a @muhlemmer what are your thoughts on this?
